File: sort/bubble_sort.py
# O(n^2)
def bubble_sort(arr):
    n = len(arr)

    for i in range(n-1):
        for j in range(n-1):
            if arr[j] > arr[j+1]:
                temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = temp
    return arr


def main():
    list = [99, 0, 5, 20, 123, 0, -1, 72, 21, 22, 13, 8, 7, 67, 29, 1, 2, 4]
    bubble_sort(list)
    print(list)

    list = [3, 9, 2, 1]
    bubble_sort(list)
    print(list)


main()

File: sort/bubble_sort.js
// define swap element fn
function swap(arr, index1, index2) {
  let temp = arr[index1];
  arr[index1] = arr[index2];
  arr[index2] = temp;
}

function BubbleSort(array) {
  let len = array.length;

  for (let j = 0; j < len - 1; j++) {
    for (let i = 0; i < len - j - 1; i++) {
      count++;
      if (array[i + 1] < array[i]) {
        swap(array, i, i + 1);
      }
    }
  }
  return array;
}

console.log(BubbleSort([8, 9, 2, 5, 1])); // [ 1, 2, 5, 8, 9 ]
